src/requestSchemas.ts

```diff
@@ -0,0 +1,201 @@
+  'expression',
+  'include-dependencies',
+  'include-components',
+  'manifest',
+  'parameters'
+];
+
+// Operation Definition: https://build.fhir.org/ig/HL7/cqf-measures/measure-repository-service.html#search
+const UNSUPPORTED_CORE_SEARCH_ARGS = [
+  'composed-of',
+  'context',
+  'context-type',
+  'context-type-quantity',
+  'context-type-value',
+  'date',
+  'depends-on',
+  'derived-from',
+  'effective',
+  'jurisdiction',
+  'predecessor',
+  'successor',
+  'topic'
+];
+
+const UNSUPPORTED_LIBRARY_SEARCH_ARGS = [...UNSUPPORTED_CORE_SEARCH_ARGS, 'content-type', 'type'];
+
+const UNSUPPORTED_MEASURE_SEARCH_ARGS = [...UNSUPPORTED_CORE_SEARCH_ARGS, 'publisher'];
+
+const customErrorMap: z.ZodErrorMap = (issue, ctx) => {
+  if (issue.code === z.ZodIssueCode.custom) {
+    if (issue.params?.type === 'NotImplemented') {
+      throw new NotImplementedError(issue.message ?? ctx.defaultError);
+    } else {
+      throw new BadRequestError(issue.message ?? ctx.defaultError, issue.params?.serverErrorCode);
+    }
+  } else if (issue.code === z.ZodIssueCode.unrecognized_keys) {
+    throw new BadRequestError(issue.message ?? ctx.defaultError, constants.ISSUE.CODE.VALUE);
+  }
+  throw new BadRequestError(issue.message ?? ctx.defaultError);
+};
+
+z.setErrorMap(customErrorMap);
+
+const hasIdentifyingInfo = (args: Record<string, any>) => args.id || args.url || args.identifier;
+
+/**
+ * Returns a function that checks if any unsupported params are present, then runs the
+ * other passed in functions in sequence. Each catchFunction is expected to check for
+ * invalid input and call ctx.addIssue if invalid input is detected. Any added issue will
+ * trigger an error to be thrown.
+ */
+export function catchInvalidParams(
+  catchFunctions: ((val: Record<string, any>, ctx: z.RefinementCtx) => void)[],
+  unsupportedArgs?: string[]
+) {
+  return (val: Record<string, any>, ctx: z.RefinementCtx) => {
+    const includedUnsupported = Object.keys(val).filter((v: string) => unsupportedArgs?.includes(v));
+    if (includedUnsupported.length > 0) {
+      ctx.addIssue({
+        code: z.ZodIssueCode.custom,
+        params: { type: 'NotImplemented' },
+        message: `Parameter(s): ${includedUnsupported.join(', ')} are not yet supported on this server`
+      });
+    } else {
+      catchFunctions.forEach(func => func(val, ctx));
+    }
+  };
+}
+
+/**
+ * Checks that either id, url, or identifier parameter is included. Adds an issue to the ctx
+ * that triggers a BadRequest to be thrown if not.
+ */
+export function catchMissingIdentifyingInfo(val: Record<string, any>, ctx: z.RefinementCtx) {
+  if (!hasIdentifyingInfo(val)) {
+    ctx.addIssue({
+      code: z.ZodIssueCode.custom,
+      params: { serverErrorCode: constants.ISSUE.CODE.REQUIRED },
+      message: 'Must provide identifying information via either id, url, or identifier parameters'
+    });
+  }
+}
+
+const stringToBool = z.enum(['true', 'false']).transform(x => x === 'true');
```

Comment:
I think I led you astray here. I tested all the edge cases, and look at this one:

```
{
  "resourceType": "Parameters",
  "parameter": [{
    "name": "id",
    "valueString": "ColorectalCancerScreeningsFHIR"
  }, {
    "name": "include-terminology",
    "valueBoolean": true
  }]
}
```

Gives the following error: `Expected 'true' | 'false', received boolean`

Issue is that `true/false` comes across as a string when encoded in the URL (reasonable), but as a boolean when in the params resource (also reasonable). We'll have to figure out some way to make both of them happy

Reply:
Could probably use a `z.union` of the current schema and `z.boolean()`?

Reply:
oof... good find. Could we just use a `z.union` here?